adapters/repos/db/shard_dimension_tracking.go

```diff
@@ -105,7 +105,11 @@ func (s *Shard) initDimensionTracking() {
 		s.publishDimensionMetrics(ctx)
 		// start tracking vector dimensions goroutine only when tracking is enabled
 		f := func() {
-			t := time.NewTicker(5 * time.Minute)
+			interval := config.DefaultTrackVectorDimensionsInterval
+			if s.index.Config.TrackVectorDimensionsInterval != 0 {
+				interval = s.index.Config.TrackVectorDimensionsInterval
```

Comment:
might not be needed as default interval is set if not specified in the env var

Reply:
It's not strictly necessary, just an added layer of protection to prevent a panic in `NewTicker` if `s.index.Config.TrackVectorDimensionsInterval` is somehow unset or set to 0